### PR TITLE
71 goprobe rotation runs into deadlock on low traffic interfaces

### LINKED ISSUE #66: goProbe not shutting down gracefully
There still seems to be an issue with the state machine somehow. I can (relatively) reliably reproduce the following behavior:

Shutdown a running instance of `goProbe` via `Ctrl-C` does not cause the state machine to terminate correctly (in fact, packets are continued to be processed - I had some debugging statements in the code that were still popping up after I sent the signal). Signal is received, but even after a while it still continues to run:
```
...
^C2023-03-06T14:29:20.333+0100	info	goProbe/goProbe.go:232	shutting down gracefully	{"app_name": "goProbe_slimcap", "app_version": "872abf81"}
...
```
I was able to abort it with a signal 6, maybe that gives a hint at where he's stuck: [abort.txt](https://github.com/els0r/goProbe/files/10898695/abort.txt)

### LINKED ISSUE #71: goProbe rotation runs into deadlock
When running a current instance of goProbe (`develop`) I can see flows sporadically not being tracked properly. Compared to a legacy version of goProbe there are several missing (although the overall numbers are not crazily far off, so it must be an issue that does not affect _all_ flows). Both queries were issued at 16:00, so the flows should have long rotated for sure:
```
# goQuery_legacy -d /usr/local/goProbe/db -i eth1.10 -f "2023-03-07 15:10:00" -c "dip = 84.46.64.49" -n 10 raw

                                                                      packets   packets             bytes      bytes       
               time    iface         sip          dip  dport  proto        in       out      %         in        out      %
  23-03-07 15:23:49  eth1.10  10.0.0.102  84.46.64.49    443    TCP    2.37 k    5.63 k   4.15  155.81 kB    7.84 MB   4.03
  23-03-07 15:38:49  eth1.10  10.0.0.102  84.46.64.49    443    TCP  488.00      1.58 k   1.07   32.62 kB    2.18 MB   1.12
  23-03-07 15:48:49  eth1.10  10.0.0.102  84.46.64.49    443    TCP   49.94 k  132.69 k  94.77    3.17 MB  184.87 MB  94.85
  23-03-07 15:53:49  eth1.10  10.0.0.102  84.46.64.49    443    TCP    4.00      8.00     0.01  264.00  B  576.00  B   0.00
                                                                                                                           
                                                                      52.81 k  139.92 k           3.35 MB  194.89 MB       
                                                                                                                    
            Totals:                                                            192.72 k                    198.25 MB  

Timespan / Interface : [2023-03-07 15:07:45, 2023-03-07 15:58:49] / eth1.10
Sorted by            : first packet time
Query stats          : 4.00   hits in 89ms
Conditions:          : dip = 84.46.64.49
```
vs.
```
# goQuery_slimcap -d /tmp/test_db -i eth1.10 -f "2023-03-07 15:10:00" -c "dip = 84.46.64.49" -n 10 raw

                                                                       packets  packets              bytes    bytes        
                 time    iface         sip          dip  dport  proto       in      out       %         in      out       %
  2023-03-07 15:25:00  eth1.10  10.0.0.102  84.46.64.49    443    TCP   2.37 k   5.63 k  100.00  155.81 kB  7.84 MB  100.00
                                                                                                                           
                                                                        2.37 k   5.63 k          155.81 kB  7.84 MB        
                                                                                                                   
              Totals:                                                            8.01 k                     7.99 MB  

Timespan / Interface : [2023-03-07 15:10:00, 2023-03-07 15:30:00] / eth1.10
Sorted by            : first packet time
Query stats          : displayed top 1 hits out of 1 in 12ms
Conditions:          : dip = 84.46.64.49
```

### PR DESCRIPTION
**Pull request by fako1024**
Closes #71 
Closes #66 

Related changes in slimcap: https://github.com/fako1024/slimcap/compare/ffc7db3..894431fb

### CODE REVIEW

File: pkg/capture/capture_mock_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	return &mockCapture{
+		maxPkts: maxPkts,
+		block:   make(chan event.EvtData, 1),
+	}
+}
+
+// NewPacket creates an empty "buffer" package to be used as destination for the NextPacketInto()
+// method. It ensures that a valid packet of appropriate structure / length is created
+func (c *mockCapture) NewPacket() capture.Packet {
+	panic("not implemented") // TODO: Implement
```
Comment (els0r) [inline pkg/capture/capture_mock_test.go:27]: Now, I'm slowly making the connection to https://github.com/fako1024/slimcap/pull/14#pullrequestreview-1357368146 😄

File: pkg/capture/capture_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+func TestHighTrafficDeadlock(t *testing.T) {
+	testDeadlock(t, -1)
+}
+
+func testDeadlock(t *testing.T, maxPkts int) {
+	ctx := context.Background()
+	mockSrc := newMockCaptureSource(maxPkts)
+	mockC := &Capture{
```
Comment (els0r) [inline pkg/capture/capture_test.go:24]: What about a `newCapture()` utility function?

File: pkg/capture/capture_test.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		},
+		rotationState: newRotationState(),
+		flowLog:       NewFlowLog(),
+		errMap:        make(map[string]int),
+		ctx:           ctx,
+		captureHandle: mockSrc,
+	}
+
+	start := time.Now()
+	time.AfterFunc(100*time.Millisecond, func() {
```
Comment (els0r) [inline pkg/capture/capture_test.go:40]: Neat use of `time.AfterFunc` +1

**Review by els0r — APPROVED** [review]
Apologies for getting to this so late. Slipped my radar.

Thank you for fixing this critical part!

File: pkg/capture/capture_mock_test.go (hunk 1 above)
Comment (fako1024) [reply pkg/capture/capture_mock_test.go:27]: Jup. When I hacked this together I immediately thought that this should be generalized / made available in `slimcap`. :rofl:

File: pkg/capture/capture_test.go (hunk 2 above)
Comment (fako1024) [reply pkg/capture/capture_test.go:24]: This will be completely replaced by the mock sources from `slimcap` in #88 , thereby making all this redundant...

**Pull request merged**